spark/src/main/java/org/apache/iceberg/actions/BaseAction.java

```diff
@@ -41,4 +50,66 @@ protected String metadataTableName(MetadataTableType type) {
       return tableName + "." + type;
     }
   }
+
+  /**
+   * Returns all the path locations of all Manifest Lists for a given table
+   * @param table the table
+   * @return the paths of the Manifest Lists
+   */
+  protected List<String> getManifestListPaths(Table table) {
+    List<String> manifestLists = Lists.newArrayList();
+    for (Snapshot snapshot : table.snapshots()) {
+      String manifestListLocation = snapshot.manifestListLocation();
+      if (manifestListLocation != null) {
+        manifestLists.add(manifestListLocation);
+      }
+    }
+    return manifestLists;
+  }
+
+  /**
+   * Returns all Metadata file paths which may not be in the current metadata. Specifically
+   * this includes "version-hint" files as well as entries in metadata.previousFiles.
+   * @param ops TableOperations for the table we will be getting paths from
+   * @return a list of paths to metadata files
+   */
+  protected List<String> getOtherMetadataFilePaths(TableOperations ops) {
```

Comment:
Same here. Since we aren't making these common utility methods, we should keep them private so we aren't committing to support them later on.